msrestazure/azure_active_directory.py

```diff
@@ -525,3 +531,32 @@ def set_token(self, response_url):
             raise_with_traceback(AuthenticationError, "", err)
         else:
             self.token = token
+
+class AdalAuthentication(Authentication):#pylint: disable=too-few-public-methods
+
+    def __init__(self, adal_method, *args, **kwargs):
+        self._adal_method = adal_method
+        self._args = args
+        self._kwargs = kwargs
+
+    def signed_session(self):
+        session = super(AdalAuthentication, self).signed_session()
+
+        import adal # Adal is not mandatory
+
+        try:
+            raw_token = self._adal_method(*self._args, **self._kwargs)
+            scheme, token = raw_token['tokenType'], raw_token['accessToken']
+        except adal.AdalError as err:
+            #pylint: disable=no-member
+            if (hasattr(err, 'error_response') and ('error_description' in err.error_response)
+                    and ('AADSTS70008:' in err.error_response['error_description'])):
+                raise Expired("Credentials have expired due to inactivity. Please run 'az login'")
```

Comment:
consider changing this error message, since this is not the CLI and they can't run az login.